libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -32,9 +34,7 @@ export class BookSearchComponent implements OnInit {
   }
 
   ngOnInit(): void {
-    this.store.select(getAllBooks).subscribe(books => {
-      this.books = books;
-    });
+    this.books$ = this.store.select(getAllBooks).pipe(takeWhile(() => this.keepSubscription));
```

Comment:
so since you changed it to using the `async` pipe in the template, there's actually no need to do the unsubscribe management anymore. Using the Async pipe makes sure that the subscription is being destroyed properly when the component gets destroyed.

If you're not able to leverage the async pipe, the patterns are usually to use `takeUntil`, like...

```
export class BookSearchComponent implements OnDestroy {
   private destroySubj = new Subject();

   ngOnInit() {
      this.store.select(getAllBooks).pipe(takeUntil(this.destroySubj)).subscribe(....)
   }

   ngOnDestroy() {
      this.destroySubj.next();
      // some people also add this.destroySubj.complete() but it is not really necessary.
   }

}
```

Using `takeUntil` might have some weird edge cases that one needs to be aware of. [Here's an article](https://medium.com/angular-in-depth/rxjs-avoiding-takeuntil-leaks-fb5182d047ef) going into the depths of that.

Another simple pattern is to create a subscription sink. Basically every `.subscribe()` returns a `Subscription` object where one can invoke `.unsubscribe()`. If you have multiple ones, you simply create an array of them and loop over them in the `ngOnDestroy` :)

Reply:
That makes sense. I remember now that the async pipe takes care of unsubscribing for you.  Thank you for the feedback.